### PR TITLE
FEM: result object refactor

### PR DESCRIPTION
**Pull request by berndhahnebach**
only changes in FEM, but quite some under the hut in the regard of result object. Lots of properties are moved to a new Python result object. Thus the C++ result object is more generic and can be used for all kind of CAE analysis. At the moment mostly needed by the CFD guys (qingfeng).

http://forum.freecadweb.org/viewtopic.php?f=18&t=20363

and lots of small fixes in FEM

cheers bernd

### CODE REVIEW

File: src/Mod/Fem/App/FemPostPipeline.cpp [hunk 1]
```diff
@@ -256,7 +258,7 @@ void FemPostPipeline::load(FemResultObject* res) {
 
     //Now copy the point data over
     //############################
-    if(res->getPropertyByName("Velocity")){
+    if(res->getPropertyByName("Velocity")){  // consider better way to detect result type, res->Type == "CfdResult"
```
Comment (luzpaz) [inline src/Mod/Fem/App/FemPostPipeline.cpp]: Maybe a **TODO:** at the beginning of the comment would be useful for future friendly searches?

File: src/Mod/Fem/_ViewProviderFemMechanicalResult.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+#*                                                                         *
+#***************************************************************************
+
+__title__ = "ViewProvider for FEM mechanical ResultObjectPython"
+__author__ = "Qingfeng Xia, Bernd Hahnebach"
+__url__ = "http://www.freecadweb.org"
+
+## @package _ViewProviderFemMechanicalResult
+#  \ingroup FEM
+
```
Comment (luzpaz) [inline src/Mod/Fem/_ViewProviderFemMechanicalResult.py]: doxygen tag can be added here
`#  \brief FreeCAD ViewProvider for mechanical ResultObjectPython in FEM workbench`

File: src/Mod/Fem/_FemMechanicalResult.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+#*                                                                         *
+#***************************************************************************
+
+__title__ = "DocumentOject Class to hold mechanical FEM results"
+__author__ = "Qingfeng Xia, Bernd Hahnebach"
+__url__ = "http://www.freecadweb.org"
+
+## @package FemMechanicalResult
+#  \ingroup FEM
+
```
Comment (luzpaz) [inline src/Mod/Fem/_FemMechanicalResult.py]: `#  \brief FreeCAD DocumentObject class to hold mechanical results in FEM workbench`

File: src/Mod/Fem/_CommandShowResult.py [hunk 4]
```diff
@@ -21,15 +21,15 @@
 # ***************************************************************************
 
 __title__ = "Command Show Result"
-__author__ = "Juergen Riegel"
+__author__ = "Juergen Riegel, Bernd Hahnebach"
 __url__ = "http://www.freecadweb.org"
 
 ## @package CommandShowResult
 #  \ingroup FEM
```
Comment (luzpaz) [inline src/Mod/Fem/_CommandShowResult.py]: #  \brief FreeCAD Command show results for FEM workbench

File: src/Mod/Fem/FemMechanicalResult.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+#*   USA                                                                   *
+#*                                                                         *
+#***************************************************************************
+
+__title__ = "module to make a mechanical FEM result object"
+__author__ = "Qingfeng Xia, Bernd Hahnebach"
+__url__ = "http://www.freecadweb.org"
+
+## \addtogroup FEM
+#  @{
```
Comment (luzpaz) [inline src/Mod/Fem/FemMechanicalResult.py]: `#  \brief FreeCAD module to make mechanical result object in FEM workbench`

File: src/Mod/Fem/App/AppFemPy.cpp [hunk 6]
```diff
@@ -263,11 +263,11 @@ class Module : public Py::ExtensionModule<Module>
         {
             App::Document* pcDoc = App::GetApplication().getActiveDocument();
             App::DocumentObject* obj = pcDoc->getObject(resName.c_str());
-            FemVTKTools::readFluidicResult(EncodedName.c_str(), obj);
+            FemVTKTools::readResult(EncodedName.c_str(), obj);
         }
         else
-            FemVTKTools::readFluidicResult(EncodedName.c_str());  //assuming activeObject can hold Result
-        
+            FemVTKTools::readResult(EncodedName.c_str());  //assuming activeObject can hold Result
```
Comment (luzpaz) [inline src/Mod/Fem/App/AppFemPy.cpp]: for uniformity sake adding a whitespace to beginning of comment 
`// assuming activeObject can hold Result`

**Comment by berndhahnebach** [comment]
done and rebased inside ...

**Comment by berndhahnebach** [comment]
Did another small fix and updated the FEM examples to the new result object. It's ok to merge from my side.

Bernd

File: src/Mod/Fem/FemMechanicalResult.py (hunk 5 above)
Comment (luzpaz) [inline src/Mod/Fem/FemMechanicalResult.py]: @berndhahnebach I apologize, I think the \brief needs to come before line 28 (https://github.com/FreeCAD/FreeCAD/pull/525/files#diff-ef0d1ff0340bc4970c0407470340592bR28)

Comment (berndhahnebach) [inline src/Mod/Fem/FemMechanicalResult.py]: done

**Comment by berndhahnebach** [comment]
there was was another doxygen tweed wront thanks to luzpaz

**Pull request merged**